### ISSUE TITLE
Surface sentry seccomp violations

### ISSUE THREAD

**Issue by iangudger** [google/gvisor#3905]
If the sentry makes a syscall which is not allowed by its seccomp policy, it is immediately killed. This is required for the security model. Unfortunately, it is difficult to discover which syscall (or argument) triggered the violation. Surfacing this information would be useful for monitoring for potential compromise and bugs in production. The current solution for debugging locally involves uncommenting a line of code and recompiling, which could also be improved upon.

One solution would be using the new seccomp notifier mechanism:
https://www.kernel.org/doc/html/latest/userspace-api/seccomp_filter.html#userspace-notification

This could be used to notify a non-sandboxed process running alongside the sentry of the details of the violation. This information could then be relayed to some monitoring or logging system.

**Comment by ianlewis** [comment]
One way I thought of to surface this info is to extend `runsc events` to provide an event when a syscall violation occurred. This would allow support for non-containerd workloads and could be picked up by the shim and relayed back to containerd etc.

It would be even better if we could surface events all the way back up to Kubernetes and surface info in the Kubernetes API.
https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/

**Comment by fvoznika** [comment]
Seccomp violations are logged to the audit log and it contains information about which syscall hit it:

```
$ grep SECCOMP /var/log/audit/audit.log
type=SECCOMP msg=audit(1599786866.478:106054): auid=183048 uid=65534 gid=65534 ses=7481 subj==unconfined pid=1901638 comm="exe" exe="/tmp/gv6/runsc" sig=31 arch=c000003e syscall=101 compat=0 ip=0x48c985 code=0x80000000
```

**Comment by iangudger** [comment]
It doesn't look like it lists the arguments which are important when doing argument-based filtering.

It also doesn't appear to indicate which sandbox the specific violation is associated with.

**Comment by ianlewis** [comment]
Yeah, not ideal but you could probably identify the sandbox from the pid
